### DISCUSSION TITLE
how to backup and restore pg schema and data

### DISCUSSION THREAD

**Discussion by waltcow** [logto-io/logto#4388] [Q&A]
how to backup and restore pg schema and data

**Comment by simeng-li** [comment]
Are you currently hosting your own instance of the OSS Logto, or are you utilizing our cloud service? If you have set up your own database, you can easily create a backup of the database by generating a dump file. This dump file serves as a snapshot of the database at a specific point in time, and you can subsequently use it to restore the database to that state whenever necessary.

**Reply by waltcow** [reply]
I host my selfhost logto not cloud version. I try to use pg_dump to backup my db, and pg_restore to recover, but role related (tenant  role) issues are resolved by myself. I think it need to be clearly documented and cli command can be more effective @simeng-li

**Comment by simeng-li** [comment]
I see, I'll put it on our feature request table for further discussion. Thanks for the report.

**Comment by simeng-li** [comment]
Hi, following a thorough review of the recent feature requests, we've determined that database backup and restore falls under infrastructure maintenance. Given our current resource constraints, we're prioritizing CIAM product features for now. We appreciate your input and will revisit this request when feasible. 
However, you are more than welcome to share a comprehensive solution with us.  Any input from the community is highly valued and encouraged.
Regards,

**Comment by waltcow** [comment]
Backup your databases
```sh
docker exec -t your-db-container pg_dumpall -c -U postgres > dump_`date +%d-%m-%Y"_"%H_%M_%S`.sql
```

Restore your databases
```sh
cat your_dump.sql | docker exec -i your-db-container psql -U postgres

```
